syntax = "proto3";

package command;

service CommandExecutor {
  rpc ExecuteCommand (CommandRequest) returns (CommandResponse);
}

service Scheduler {
  rpc ScheduleCommand (SchedulerRequest) returns (SchedulerResponse);
}

message CommandRequest {
  string command = 1;
}

message CommandResponse {
  string result = 1;
  int32 status_code = 2;
}

message SchedulerRequest {
  string worker_id = 1;
  string command = 2;
}

message SchedulerResponse {
  string result = 1;
  int32 status_code = 2;
}
